### PR TITLE
LayoutTypes: Cleanup old types that are no longer needed 

### PR DESCRIPTION
**Pull request by torkelo**
* Simplify types now that the union type SceneObjectState is no longer needed 
* Removes SceneLayoutChildState

### CODE REVIEW

File: packages/scenes/src/core/types.ts [hunk 1]
```diff
@@ -60,7 +54,7 @@ export interface SceneDataState extends SceneObjectStatePlain {
   data?: PanelData;
 }
 
-export interface SceneObject<TState extends SceneObjectState = SceneObjectState> {
+export interface SceneObject<TState extends SceneObjectStatePlain = SceneObjectStatePlain> {
```
Comment (torkelo) [inline packages/scenes/src/core/types.ts:57]: think we should probably take the time to rename SceneObjectStatePlain to SceneObjectState or SceneObjectStateBase

File: packages/scenes/src/core/types.ts [hunk 2]
```diff
@@ -21,16 +21,10 @@ export interface SceneObjectStatePlain {
...
 
-export interface SceneLayoutChildState extends SceneObjectStatePlain {
-  placement?: SceneLayoutChildOptions;
-}
-
 export interface SceneLayoutItemState extends SceneObjectStatePlain {
   body: SceneObject | undefined;
 }
 
-export type SceneObjectState = SceneObjectStatePlain | SceneLayoutState | SceneLayoutChildState;
```
Comment (torkelo) [inline packages/scenes/src/core/types.ts:32]: this is the main thing that I don't think we need anymore (this union type was only used from NestedScene in a bit of code that no longer works (where it tries to remove itself from it's parent children array, but it's parent never has a children array in the current layouts)

Comment (dprokop) [reply packages/scenes/src/core/types.ts:32]: > this union type was only used from NestedScene in a bit of code that no longer works (where it tries to remove itself from it's parent children array, but it's parent never has a children array in the current layouts)

It does work, added in https://github.com/grafana/scenes/pull/97/commits/fa5b1412e9abbd5d75dcdfa4828813882880f072#diff-a02fdda4e1d01bbcde83f440d8f484157069ab91780e41555f9d4f7d76a467e0 and the updated in https://github.com/grafana/scenes/pull/97/commits/f1eabcaa71580dd18b92de6ef5cd5d2de001cfd4#diff-a02fdda4e1d01bbcde83f440d8f484157069ab91780e41555f9d4f7d76a467e0

File: packages/scenes/src/core/types.ts (hunk 1 above)
Comment (dprokop) [reply packages/scenes/src/core/types.ts:57]: Agree

**Review by dprokop — APPROVED** [review]

**Comment by torkelo** [comment]
@dprokop will rename SceneObjectStatePlain to SceneObjectState in another PR (as it will touch so many files, and warrants a release notes entry)

**Pull request merged**